storage-proofs-update/tests/circuit_poseidon.rs

```diff
@@ -0,0 +1,163 @@
+        .map(|_| TreeRDomain::random(&mut rng))
+        .collect();
+    let tree_d_new = common::create_tree::<TreeR>(&labels_d_new, tmp_path, "tree-d-new");
+    let comm_d_new = tree_d_new.root();
+
+    // `phi = H(comm_d_new || comm_r_old)`
+    let phi = phi(&comm_d_new, &comm_r_old);
+
+    // Encode `labels_d_new` into `labels_r_new` and create TreeRNew.
+    let labels_r_new = common::encode_new_replica(&labels_r_old, &labels_d_new, &phi, h);
+    let tree_r_new = common::create_tree::<TreeR>(&labels_r_new, tmp_path, "tree-r-new");
+    let root_r_new = tree_r_new.root();
+    let comm_r_new = <TreeRHasher as Hasher>::Function::hash2(&comm_c, &root_r_new);
+
+    let pub_params = PublicParams::from_sector_size(sector_bytes as u64);
+
+    {
+        // Generate vanilla-proof.
+        let challenge_proofs: Vec<circuit_poseidon::ChallengeProofVanilla<TreeR>> =
+            Challenges::new_poseidon(sector_nodes, comm_r_new)
+                .enumerate()
+                .take(pub_params.challenge_count * pub_params.partition_count)
+                .map(|(i, c)| {
+                    let c = c as usize;
+                    let proof_r_old = tree_r_old.gen_proof(c).unwrap_or_else(|_| {
+                        panic!("failed to generate `proof_r_old` for c_{}={}", i, c)
+                    });
+                    let proof_d_new = tree_d_new.gen_proof(c).unwrap_or_else(|_| {
+                        panic!("failed to generate `proof_d_new` for c_{}={}", i, c)
+                    });
+                    let proof_r_new = tree_r_new.gen_proof(c).unwrap_or_else(|_| {
+                        panic!("failed to generate `proof_r_new` for c_{}={}", i, c)
+                    });
+
+                    circuit_poseidon::ChallengeProofVanilla {
+                        proof_r_old,
+                        proof_d_new,
+                        proof_r_new,
+                    }
+                })
+                .collect();
+        assert_eq!(
+            challenge_proofs.len(),
+            pub_params.challenge_count * pub_params.partition_count
+        );
+        // Create circuit.
+        let pub_inputs = circuit_poseidon::PublicInputs::new(
+            sector_nodes,
+            h,
+            comm_r_old,
+            comm_d_new,
+            comm_r_new,
+        );
+
+        let pub_inputs_vec = pub_inputs.to_vec();
+
+        let priv_inputs = circuit_poseidon::PrivateInputs::new(comm_c, &challenge_proofs);
+
+        let circuit = circuit_poseidon::EmptySectorUpdateCircuit::<TreeR> {
+            pub_params,
+            pub_inputs,
+            priv_inputs,
+        };
+
+        let mut cs = TestConstraintSystem::<Fr>::new();
+        circuit.synthesize(&mut cs).expect("failed to synthesize");
+        assert!(cs.is_satisfied());
+        assert!(cs.verify(&pub_inputs_vec));
+        assert_eq!(cs.num_constraints(), constraints_expected);
+    }
+}
+
+#[test]
+#[cfg(feature = "isolated-testing")]
```

Comment:
I'll have to check what CI is doing, but since there's already a test with this name using the `isolated-testing` flag, I would think that both tests are getting run at the same time, which may result in CI RAM issues.  Once verified, we may need to update the test names and add entries to run them in the .circleci/config

Reply:
Ah, nice, since they are separate tests, they appear to also be run independently on CI.